packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,485 @@
+/// as per iOS behavior.
+///
+/// If a [CupertinoTableView] does not contain enough cells to fill the viewport
+/// then extra dividers are rendered beyond the last cell to the end of the
+/// viewport. This extra divider rendering extends into overscroll space at the
+/// end, too. The gap between extra dividers is equal to the height of the
+/// last cell in the [CupertinoTableView].
+///
+/// #Styles
+///
+/// A [CupertinoTableView] may be rendered either in a "plain" style or a
+/// "grouped" style.
+///
+/// A plain [CupertinoTableView] renders a single set of cells without any
+/// divisions between cells.
+///
+/// A grouped [CupertinoTableView] subdivides cells into sections. Sections
+/// are rendered with gaps between them and optional headers before each section.
+///
+/// #Section Index
+///
+/// If a [CupertinoTableView] is in a grouped style, it can optionally display
+/// a section "index". This index is a thin strip that appears on the right side
+/// of the [CupertinoTableView], displays titles for each section in the
+/// [CupertinoTableView], and allows users to tap on a given section title to
+/// automatically scroll to that section.
+///
+/// See also:
+///
+///  * [CupertinoTableViewCell], which renders an iOS-style table view cell that
+///    is intended to appear within a [CupertinoTableView].
+///  * [ListView], which renders a style agnostic list.
+class CupertinoTableView extends StatefulWidget {
+  /// Creates a plain-style [CupertinoTableView] with an explicit list of cells.
+  const CupertinoTableView.plain({
+    this.scrollController,
+    this.backgroundColor = CupertinoColors.white,
+    this.plainChildren = const <Widget>[],
+  }) : plainChildrenBuilder = null,
+       plainChildCount = null;
+
+  /// Creates a plain-style [CupertinoTableView] with a lazily built list of cells.
+  const CupertinoTableView.plainBuilder({
+    this.scrollController,
+    this.backgroundColor = CupertinoColors.white,
+    this.plainChildCount,
+    @required this.plainChildrenBuilder,
+  }) : plainChildren = null;
+
+  /// ScrollController that controls the scroll position of this [CupertinoTableView].
+  final ScrollController scrollController;
+
+  /// The table's background color beneath its cells.
+  final Color backgroundColor;
+
+  /// Explicit list of child cell widgets. If [plainChildren] is provided then
+  /// [plainChildrenBuilder] must be null.
+  final List<Widget> plainChildren;
+
+  /// Lazily built list of child cell widgets. If [plainChildrenBuilder] is
+  /// provided then [plainChildren] must be null.
+  final IndexedWidgetBuilder plainChildrenBuilder;
+
+  /// The number of children that can be built with [plainChildrenBuilder].
+  ///
+  /// The [plainChildCount] is optional. If it's provided then only a subset
+  /// of the plain children will be built at a time, based on viewport size. If
+  /// [plainChildCount] is not provided, then all plain children will be
```

Comment:
Didn't read the rest yet but this seems surprising.

Reply:
![image](https://user-images.githubusercontent.com/7259036/48881131-824b1180-edc8-11e8-90ec-3c88db78e84d.png)